providers/idle_timeout.rb

```diff
@@ -0,0 +1,120 @@
+#
+# Cookbook Name:: gecos-ws-mgmt
+# Provider:: idle_timeout
+#
+# Copyright 2013, Junta de Andalucia
+# http://www.juntadeandalucia.es/
+#
+# All rights reserved - EUPL License V 1.1
+# http://www.osor.eu/eupl
+#
+
+
+
+action :setup do
+
+  begin
+
+    if new_resource.support_os.include?($gecos_os)
+
+      package "zenity" do
+        action :nothing
+      end.run_action(:install)
+     
+      users = new_resource.users
+
+      users.each_key do |user_key|
+        nameuser = user_key
+        username = nameuser.gsub('###','.')
+        user = users[user_key]
+        gid = Etc.getpwnam(username).gid
+
+        Chef::Log.debug("idle_timeout ::: user = #{user}")
+        Chef::Log.debug("idle_timeout ::: username = #{username}")
+
+        autostart = ::File.expand_path("~#{username}/.config/autostart")
+        Chef::Log.debug("idle_timeout ::: autostart file = #{autostart}")
+
+        if user.idle_enabled
+
+          package 'xautolock' do
```

Comment:
Are you installing and removing xautolock packages for every user? 

I am afraid final xautolock package state will depend of the policy parameters of the last user

Reply:
You're right! Corrected.